engine/security/trust/content_trust.md

```diff
@@ -30,6 +30,11 @@ ensure that the images they use are signed. Publishers and consumers can be
 individuals alone or in organizations. Docker's content trust supports users and
 automated processes such as builds.
 
+When you enable content trust, signing occurs on the client before push and
```

Comment:
Technically signing happens after push. This ensures we sign the thing that got accepted by the registry.

Reply:
So we allow the image to be pushed, then the registry tell us that it got pushed, then the client signs the **local**  notion of it and re-pushes that? Could it change after pushing but before signing such that the client and server have different things they think are the same?

Reply:
It could change, but what the server thinks is irrelevant. Notary signing is designed such that a completely malicious registry _still_ can't make me trust something I shouldn't. The main benefit of signing after pushing is in case something goes wrong in the push.